### ISSUE TITLE
[iOS/macOS] HEVC+Alpha - alpha channel is ignored

### ISSUE THREAD

**Issue by ZimM-LostPolygon** [RenderHeads/UnityPlugin-AVProVideo#1070]
**Describe the issue**
HEVC+Alpha seems to not be working on iOS/macOS. The video plays fine, but the alpha channel is completely ignored. Same video plays correctly with alpha channel in Safari. Same thing happens with any other HEVC+Alpha videos I've tried.

**Your Setup (please complete the following information):**
 - Unity version: 2020.3.20
 - AVPro Video version (number and edition (trial/core/ultra/enterprise)): 2.2.2 Ultra
 - Operating system version: macOS 12.1, iOS 15.1
 - Device model: Mac Mini M1
 - Video specs (resolution, frame-rate, codec, file size): 360x480, 60 FPS, HEVC+Alpha, 2.3 MB

**To Reproduce**
1. Attempt to play the attached video on macOS or iOS using AVProVideo.
2. Observe the lack of alpha channel.

**Screenshots**
In Safari (you can see the alpha channel is applied)
![image](https://user-images.githubusercontent.com/928550/150143119-476804b7-ab00-44af-9d47-1d7391b46513.png)

In Unity (no alpha channel):
![image](https://user-images.githubusercontent.com/928550/150143229-d10230e5-288a-4331-a663-a226dea6f246.png)

**Videos**
Original video:
https://trac.ffmpeg.org/raw-attachment/ticket/7965/puppets_with_alpha_hevc.mov

**Comment by ZimM-LostPolygon** [comment]
@MorrisRH I'm not sure about the "enhancement" label 😀 This is clearly a bug. It's an important feature that we need, and it's listed as something AVProVideo supports right on top:
https://www.renderheads.com/content/docs/AVProVideo/articles/feature-transparency.html

**Comment by MorrisRH** [comment]
Not ignored exactly, just that support for HEVC+Alpha has not been implemented yet.
We do offer packed alpha modes which you could use to work around this for now, details are [here](https://renderheads.com/content/docs/AVProVideo/articles/component-media-player.html#source) and instructions on how to encode are available [here](https://renderheads.com/content/docs/AVProVideo/articles/encoding-notes.html#transparency-packing).

**Comment by ZimM-LostPolygon** [comment]
@MorrisRH I'm not sure about the "enhancement" label 😀 This is clearly a bug. It's an important feature that we need, and it's listed as something AVProVideo supports right on top:
renderheads.com/content/docs/AVProVideo/articles/feature-transparency.html

**Comment by ZimM-LostPolygon** [comment]
That contradicts the documentation. I would expect HEVC+Alpha to work on iOS and macOS platforms, but it doesn't.

**Comment by MorrisRH** [comment]
Yeah, sorry about that... clearly the docs are wrong here. We will look to get this implemented ASAP though.

**Comment by ZimM-LostPolygon** [comment]
Ohh... That sucks for us, because it was one of the reasons for buying AVProVideo. Alpha packing is less than optimal, so we wanted to leverage "real" alpha channel video on at least iOS.

**Comment by ZimM-LostPolygon** [comment]
Thank you! 

On a related subject - we don't know upfront whether the video would have an alpha channel or not. We need to know because we have to do scene compositing a little bit different in both cases. However, I couldn't find any API that exposes that metadata. I would expect `BaseMediaPlayer.GetTextureTransparency()` to return `Opaque` or `Transparent` depending on if the video has alpha channel or not, but seems like all implementations of `GetTextureTransparency` use `MediaHints`, which are, it seems, are always set in Unity Editor.

TL;DR: Is there any dynamic way to know if a currently open video has alpha channel or not?

**Comment by MorrisRH** [comment]
Support for HEVC+Alpha has been added and will be available in the next release. I've also made ```ITextureProducer.GetTextureTransparency()``` behave correctly with videos that contain alpha.

**Comment by ZimM-LostPolygon** [comment]
Awesomeness!!

**Comment by ZimM-LostPolygon** [comment]
@MorrisRH Unfortunately, it doesn't seem to be working even with the sample video attached to this issue:
![изображение](https://user-images.githubusercontent.com/928550/152969560-5f19f7cc-dbff-4094-a736-3245dfe24bf0.png)
Am I doing something wrong? Does something else needs to be changed?

**Comment by MorrisRH** [comment]
I tested it using our uGUI component where it was working fine. It could be that your use case isn't being met by one of our components. Could you give more details about how your scene is setup?

**Comment by ZimM-LostPolygon** [comment]
Scratch all that, I'm an dummy. I've been playing a very similarly named, but different file. It actually works! Thank you.
